sentry-android-core/src/main/java/io/sentry/android/core/internal/util/ScreenshotUtils.java

```diff
@@ -0,0 +1,73 @@
+package io.sentry.android.core.internal.util;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.os.Build;
+import android.view.View;
+import androidx.annotation.Nullable;
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.BuildInfoProvider;
+import java.io.ByteArrayOutputStream;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public class ScreenshotUtils {
+  public static @Nullable byte[] takeScreenshot(
+      final @NotNull Activity activity,
+      final @NotNull ILogger logger,
+      final @NotNull BuildInfoProvider buildInfoProvider) {
+    if (!isActivityValid(activity, buildInfoProvider)
+        || activity.getWindow() == null
+        || activity.getWindow().getDecorView() == null
+        || activity.getWindow().getDecorView().getRootView() == null) {
+      logger.log(SentryLevel.DEBUG, "Activity isn't valid, not taking screenshot.");
+      return null;
+    }
+
+    final View view = activity.getWindow().getDecorView().getRootView();
+    if (view.getWidth() <= 0 || view.getHeight() <= 0) {
+      logger.log(SentryLevel.DEBUG, "View's width and height is zeroed, not taking screenshot.");
+      return null;
+    }
+
+    try {
+      // ARGB_8888 -> This configuration is very flexible and offers the best quality
+      final Bitmap bitmap =
+          Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.ARGB_8888);
+
+      final Canvas canvas = new Canvas(bitmap);
+      view.draw(canvas);
+
+      final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
```

Comment:
M: Likely an improvement I forgot to do before:
`ByteArrayOutputStream` can be created within a try with resources so it auto closes the stream, right now if it fails for some reason, the stream is kept open.
Example:
`try (final ByteArrayOutputStream stream = new ByteArrayOutputStream();) {}`